language-bindings/python/src/wamr/wamrapi/wamr.py

```diff
@@ -129,21 +179,53 @@ def _create_module_inst(self, module: Module, stack_size: int, heap_size: int) -
         return module_inst
 
 
+id_to_exec_env_mapping = {}
+
+
 class ExecEnv:
-    def __init__(self, module_inst: Instance, stack_size: int = 65536):
+    def __init__(self, module_inst: Instance | None = None, stack_size: int = 65536):
```

Comment:
I am not sure `module_inst` is `None`able. It should be quite critical. Please make more tests